src/stores/room-list/algorithms/Algorithm.ts

```diff
@@ -208,6 +212,12 @@ export class Algorithm extends EventEmitter {
             position: position,
             tag: tag,
         };
+
+        // We update the filtered rooms just in case, as otherwise users will end up visiting
+        // a room while filtering and it'll disappear. We don't update the filter earlier in
+        // this function simply because we don't have to.
+        this.recalculateFilteredRoomsForTag(tag);
+        if (lastStickyRoom && lastStickyRoom.tag !== tag) this.recalculateFilteredRoomsForTag(tag);
```

Comment:
Did you mean to add the unconditional call to `recalculateFilteredRoomsForTag` as well?

Reply:
to the first one? No, it is null guarded on line 172

Reply:
Right, but you call `this.recalculateFilteredRoomsForTag(tag);` on line 219 and then call it again conditionally on the next line.

Reply:
ohhhhhhhhhhhhhhhhhhhhhhhh that should be the last tag